WebHostLib/upload.py

```diff
@@ -11,18 +11,30 @@
 from markupsafe import Markup
 from pony.orm import commit, flush, select, rollback
 from pony.orm.core import TransactionIntegrityError
+import schema
 
 import MultiServer
 from NetUtils import SlotType
 from Utils import VersionException, __version__
+from worlds import GamesPackage
 from worlds.Files import AutoPatchRegister
+from worlds.AutoWorld import data_package_checksum
 from . import app
 from .models import Seed, Room, Slot, GameDataPackage
 
 banned_extensions = (".sfc", ".z64", ".n64", ".nes", ".smc", ".sms", ".gb", ".gbc", ".gba")
 allowed_options_extensions = (".yaml", ".json", ".yml", ".txt", ".zip")
 allowed_generation_extensions = (".archipelago", ".zip")
 
+games_package_schema = schema.Schema({
```

Comment:
- did you look into validating via TypedDict instead of Schema? (we have a TypedDict in `worlds/__init__.py`. I read up on that and the frameworks that support(ed) that were massive, so I gave up on that idea, but I may have missed something)
- should we maybe move the Schema to `worlds/__init__.py` so it's local to the TypedDict?

Reply:
I considered that we already have a typed dict, but also am not a aware of a small and quick way of validating it. But we do already have  schema, so I went with that.

Second, maybe, I considered that too, I ended up moving it to webhost to keep init time low as we avoid a schema import if we don't need it. But I also didn't measure how long the module needs, so this  may  be over-engineered.